src/couchapps/WorkQueue/lists/workRestrictions.js

```diff
@@ -65,70 +66,79 @@ function(head, req) {
             continue;
         }
 
-        // Don't check anything if trustSitelists is enabled
-        if (ele['NoLocationUpdate']) {
-            // subtract element jobs from site resources
-            if (first !== true) {
-                send(",");
-            }
-            send(toJSON(row["doc"])); // need whole document, id etc...
-            first = false; // from now on prepend "," to output
-            continue; // we have work, move to the next element
-        }
-
         for (var site in resources) {
 
             // skip if in blacklist
             if (ele["SiteBlacklist"].indexOf(site) != -1) {
                 continue;
             }
+
             //skip if not in whitelist
             if (ele["SiteWhitelist"].length != 0 && ele["SiteWhitelist"].indexOf(site) === -1) {
                 continue;
             }
-            //skip if parent processing flag is set and parent block is not in the site.
-            //all the parent block has to be in the same site
-            noParentSite = false;
-            if (ele["ParentFlag"]) {
-                for (block in ele["ParentData"]) {
-                    if (ele["ParentData"][block].indexOf(site) === -1) {
-                        noParentSite = true;
+
+            // Don't check anything if trustSitelists AND trustPUSitelists is enabled
+            if ('NoLocationUpdate' in ele && ele['NoLocationUpdate'] && ele["SiteWhitelist"].indexOf(site) !== -1) {
```

Comment:
In javascript, you don't have to check - 'NoLocationUpdate' in ele, 
ele[ 'NoLocationUpdate'] return undefined value which is considered false.
So following checking will be enough.

 ele['NoLocationUpdate'] 

But if you make more clear condition checking,  you can do 

 ele['NoLocationUpdate']  === true